infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -248,10 +248,12 @@ public synchronized void handleMessageLayoutPrepare(CorfuPayloadMsg<LayoutPrepar
             r.sendResponse(ctx, msg, CorfuMsgType.LAYOUT_PREPARE_REJECT.payloadMsg(new
                     LayoutPrepareResponse(phase1Rank.getRank(), proposedLayout)));
         } else {
+            // Return the layout with the highest rank proposed before.
+            Rank highestProposedRank = proposedLayout == null ? prepareRank : getPhase1Rank();
```

Comment:
this is wrong!

1. if there is no proposedLayout, return -1. 
(the proposed Layout will be null in this case, but don't rely on this, it is bug prone). 

2. if there is a proposeLayout, return getPhase2Rank().  
   (phase1 might have increased since the last accepted propose, without actually getting any proposed layout!)

Reply:
Ok done.